hack/generator/pkg/astmodel/definition.go

```diff
@@ -13,8 +13,19 @@ type HasImports interface {
 	RequiredImports() []PackageReference
 }
 
+// ReferenceChecker describes an entity that can check if it references a given type
+type ReferenceChecker interface {
+	// Does this include any direct references to the given Type?
+	// "Direct" means we don't walk into any StructReferences (nor could we with these arguments),
+	// but we do walk into included StructTypes.
+	References(t Type) bool
```

Comment:
As discussed, I think this needs to be renamed to `IsReferenceBy()` to make it clear both that this is a boolean test and the direction of that test.

Reply:
The direction is `References` not `IsReferencedBy`, since we only recurse through the receiver side, not the argument side.

Reply:
Clearly we're each interpreting the direction of `References` and `IsReferencedBy` differently. 😁 

I think this proves we need a better name for the method, though I'm not sure quite what that is ...

Reply:
it’s the same direction as the normal verb isn’t it? `x.References(y)`

Reply:
It's possible to read `x.References(y)` as a mutator that establishes the relationship that `x has a reference to y`, verses reading it as a test for whether that relationship already exists.

I know I'm splitting some mighty fine hairs here, let's just say that I'm scarred by experiences with developers who just don't care about the semantics of a name.

Taking into account the direction - how about `HasReferenceTo()` and also making the check bidirectional when allocating types to files?